heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraHeronShellController.java

```diff
@@ -0,0 +1,119 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.scheduler.aurora;
+
+import java.net.HttpURLConnection;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import com.twitter.heron.proto.system.PhysicalPlans.StMgr;
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.ConfigLoader;
+import com.twitter.heron.spi.common.Context;
+import com.twitter.heron.spi.packing.PackingPlan;
+import com.twitter.heron.spi.statemgr.IStateManager;
+import com.twitter.heron.spi.statemgr.SchedulerStateManagerAdaptor;
+import com.twitter.heron.spi.utils.NetworkUtils;
+import com.twitter.heron.spi.utils.ReflectionUtils;
+
+/**
+ * Implementation of AuroraController that is a wrapper of AuroraCLIController
+ * The difference is: the `restart` method implementation is changed to heron-shell
+ */
+class AuroraHeronShellController implements AuroraController {
+  private static final Logger LOG = Logger.getLogger(AuroraHeronShellController.class.getName());
+
+  private final String topologyName;
+  private final AuroraCLIController cliController;
+  private SchedulerStateManagerAdaptor stateMgrAdaptor;
+
+  AuroraHeronShellController(
+      String jobName,
+      String cluster,
+      String role,
+      String env,
+      String auroraFilename,
+      boolean isVerbose) {
+    this.topologyName = jobName;
+    this.cliController = new AuroraCLIController(
+        jobName, cluster, role, env, auroraFilename, isVerbose);
+
+    stateMgrAdaptor = null;
+    Config config =
+        Config.toClusterMode(Config.newBuilder().putAll(ConfigLoader.loadClusterConfig()).build());
+    String stateMgrClass = Context.stateManagerClass(config);
+    IStateManager stateMgr = null;
+    try {
+      stateMgr = ReflectionUtils.newInstance(stateMgrClass);
+
+      stateMgr.initialize(config);
+      stateMgrAdaptor = new SchedulerStateManagerAdaptor(stateMgr, 5000);
+    } catch (ClassNotFoundException e) {
+      LOG.severe(e.getMessage());
+    } catch (InstantiationException e) {
+      LOG.severe(e.getMessage());
+    } catch (IllegalAccessException e) {
+      LOG.severe(e.getMessage());
+    }
```

Comment:
Need we throw an exception if we could not initialize `stateMgr` and `stateMgrAdaptor`? Otherwise, the `AuroraHeronShellController` is in a wrong state.

Reply:
Eventually any exception has to be caught before AuroraScheduler.initialize(). If the exception is not caught here, where shall we catch them?

Reply:
Bubble up a RuntimeException to external logic. Otherwise, the program will execute with a bad state but it is no way to execute with a bad state.

Reply:
done: throw to external logic